billingoptions.php

```diff
@@ -119,18 +128,22 @@ function InputChecker($data)
       $Q .= "'" . $bill_p2_f . "'";
       $Q .= ",bill_other=";
       $Q .= "'" . $bill_other_f . "'";
+      $Q .= ",no_comment=";
```

Comment:
While we are here, maybe we could clean up these SQL statements a little bit and use string interpolation and heredoc to make it easier to read.
* string interpolation https://www.oreilly.com/library/view/programming-php-3rd/9781449361068/ch04s01.html
* heredocs https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc
* an example `members-list.php:241` or https://github.com/tim-hogan/glidingops/blob/856296041880f6fedde853b65e0418ea0d12d57b/members-list.php#L244

What do you think?

Reply:
@brunotag could you please upload a screenshot of the `TREASURERS REPORT` for a month in which you have some flights charged with GWR? I want to see in which section of the report do these flights get listed.

Reply:
I can try, but according to what Brian said, that's out of scope.

Reply:
![image](https://user-images.githubusercontent.com/1215492/65593306-4a15f000-dfe4-11e9-94a2-0e9da036b33e.png)
They fall into the Trial flights.

That's probably because the Treasurer.php page assumes that if no PIC or P2 is charged AND it is not a NoCharge flight, it must be a Trial ...

These bad things happen when a UI makes specific assumptions about generic data :-(

Reply:
Currently, though, they fall into NO CHARGE FLIGHTS... so it's not a regression.

It's as bad as.. 

![image](https://user-images.githubusercontent.com/1215492/65593538-cc9eaf80-dfe4-11e9-96fa-c23b62294b0f.png)

Reply:
I see now why you are thinking about a "Charge 3rd Party" kind of thing.

You could use that to dynamically build a report with a section, and a calculated total, for each 3rd party

Reply:
I'll ping the treasurer and I'll ask him how he currently uses the report to charge GWR.

Reply:
Ok, I forgot about 50/50. You are right, trying to model these charge types with a single column is too much of a change now. Let us drop the idea of one column.

However, I still think that `no_comment` is not appropriate. All the other checkboxes indicate who is the charge targeted to and this one should do the same. The fact that charging someone outside the system does not need to have a comment, doesn’t mean it has to be modelled as a no_comment field.

Furthermore, I think it is a fairly simple change to update the treasure’s report and add one section for each of this special charges (however you decide to call them in the end). The name of the billing option can make the title of the section etc.

Reply:
Ok, let's put it this way: we have some logic that asks for "details" (in the comment field) for some billing modes, and we want to go around it.

We could turn the boolean around and change it to something like "Requires Billing Details" instead of "No Comment", and change the logic to check for that.

But that wouldn't be easy to express anyway, semantically, because currently our check is quite convoluted: it asks for comments only if all three Charge PIC, Charge P2 and Charge Other are false (zero) and the flight is a glider flight type.
How do you express that neatly?
I don't know.

Anyway, those conditions are true for the three types of Trial we have and for NoCharge (see the screenshot below). 
I don't wanna change that because I don't know what else it could break. 

I am aware that it is hacky, but the only idea I have is to add a field that says "do not ask for details for this kind of billing option", and I would call it "No Comment", and that works, and nobody would see that except the admin.

So, unless you can find an alternative (involving an equivalent amount of work, not turning the code around), I'll go with that.

Reply:
We could change it, from "No Comment" to "Requires Comment", and make that true for the Trials and for the No Charge billing options.

That is a bit less hacky.